sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -72,26 +74,31 @@ namespace Azure { namespace Core { namespace Http {
 
   struct RetryOptions
   {
-    int16_t MaxRetries = 5;
-    int32_t RetryDelayMsec = 500;
+    int32_t MaxRetries = 3;
+
+    std::chrono::duration<int64_t, std::milli> RetryDelay = std::chrono::seconds(4);
+    decltype(RetryDelay) MaxRetryDelay = std::chrono::minutes(2);
+
+    std::vector<HttpStatusCode> StatusCodes{
+        HttpStatusCode::RequestTimeout,
+        HttpStatusCode::InternalServerError,
+        HttpStatusCode::BadGateway,
+        HttpStatusCode::ServiceUnavailable,
+        HttpStatusCode::GatewayTimeout,
+    };
   };
 
   class RetryPolicy : public HttpPolicy {
   private:
     RetryOptions m_retryOptions;
 
   public:
-    explicit RetryPolicy(RetryOptions options) : m_retryOptions(options) {}
+    explicit RetryPolicy(RetryOptions const& options) : m_retryOptions(options) {}
```

Comment:
Does this make a deep copy of the vector? If need it to if it doesn't

Reply:
It does, because you may create a temporary object. I expect these vectors to be small, there should be no problem. Not to say it is not possible to optimize, but I don't see this becoming a bottleneck.
BTW, for the perf, we can put these codes in a hashset instead of a vector, but again, I don't think we should do it now.

Reply:
I'm not worried about the perf at all. I'm worried about the thread safety. I don't want one thread changing the vector while another thread is scanning the vector.

Reply:
THen there should be no problem - we make the copy. We take the reference, but have the copy as a the data member - and copy it the first thing we get into constructor.

Reply:
In other words, your feedback is addressed - we DO copy.

Reply:
I don't feel current implementation to be better than what is written below, but it's nitpicking:
```
explicit RetryPolicy(RetryOptions options) : m_retryOptions(std::move(options)) {}
```

Reply:
Thank you, I'll do this.

Reply:
Done.